### PR TITLE
build: resync enhancements

### PR DESCRIPTION
**Pull request by jeff-mccoy**
This PR simplifies the resync interval logic to prevent possible race conditions with JS timers by moving the resync logic to an out-of-band (JS `setInterval`) process instead of `setTimeout`.

### CODE REVIEW

File: src/fluent/watch.ts [hunk 1]
```diff
@@ -202,6 +216,19 @@ export class Watcher<T extends GenericClass> {
...
+    // Create a new agent with keep-alive and the TLS options
+    opts.agent = new Agent({
+      keepAlive: true,
+      keepAliveMsecs: 30 * 1000,
+      timeout: 30 * 1000,
+      ca,
+      cert,
+      key,
+    });
+
```
Comment (cmwylie19) [inline src/fluent/watch.ts:231]:
```suggestion
```

File: src/fluent/watch.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import byline from "byline";
 import { createHash } from "crypto";
 import { EventEmitter } from "events";
+import { Agent } from "https";
```
Comment (cmwylie19) [inline src/fluent/watch.ts:7]:
```suggestion
```

**Comment by cmwylie19** [comment]
i like the faster feedback loop. This is looking good

**Review by cmwylie19 — APPROVED** [review]

**Review by schaeferka — APPROVED** [review]

**Review by btlghrants — APPROVED** [review]
A move in the right direction -- I think the stable resync windows'll help make troubleshooting just  little bit easier.

**Pull request merged**